game/src/core/components/DrawComponent.java

```diff
@@ -66,19 +66,32 @@ public class DrawComponent extends Component {
      * @throws IOException if the given path does not exist
      * @see Animation
      */
-    public DrawComponent(Entity entity, String path) throws IOException {
+    public DrawComponent(final Entity entity, final String path) throws IOException {
         super(entity);
         // fetch available animations
-        ClassLoader classLoader = getClass().getClassLoader();
-        File directory = new File(classLoader.getResource(path).getFile());
-        if (!directory.exists() || !directory.isDirectory()) {
-            throw new FileNotFoundException("Path " + path + " not found.");
+        try {
+            ClassLoader classLoader = getClass().getClassLoader();
+            File directory = new File(classLoader.getResource(path).getFile());
+            animationMap =
+                    Arrays.stream(directory.listFiles())
+                            .filter(File::isDirectory)
+                            .collect(Collectors.toMap(File::getName, Animation::of));
+            currentAnimation(CoreAnimations.IDLE_LEFT);
+        } catch (NullPointerException np) {
+            /* The component gets registered at the entity in super().
+               This means that if the files can't be loaded, the component is considered "defective" and will likely throw exceptions.
+               For that reason, we remove the defective component from the entity.
+            */
+            entity.removeComponent(DrawComponent.class);
+            /* We convert the "NullPointerException" to a "FileNotFoundException" because the only
+               reason for a NullPointerException is if the directory does not exist.
+            */
```

Comment:
```suggestion
            // We convert the "NullPointerException" to a "FileNotFoundException" because the only
            // reason for a NullPointerException is if the directory does not exist.
```